### PR TITLE
New version of the graphviz package.

### PR DESCRIPTION
**Pull request by mpan322**
Latest Version of the Graphviz Package: 
 -  Sorry about the merge issues I have a comment below about it. Shold be able to just chuck out the old code. My bad for not managing the changing properly with git.
 - Need to update the PackageInfo.g and some docs but functionality should be working.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
I noticed while trying to implement some existing functions using graphviz that if I add a subgraph to a digraph, then the edges are incorrectly written as `--` in the subgraph (which also makes the output invalid `dot` code).

File: gap/dot.gi [hunk 1]
```diff
@@ -572,7 +579,12 @@ InstallMethod(GraphvizSetAttr, "for a graphviz graph, object and object",
 [IsGVGraph, IsObject, IsObject], 
 function(x, name, value)
   local attrs, string;
-  
+
+  # display warning if not known attribute
+  if not name in GRAPHVIZ_KNOWN_ATTRS then
+    Print(StringFormatted("[WARNING] Unkown attribute {}\n", name));
```
Comment (james-d-mitchell) [inline gap/dot.gi:585]:
```suggestion
    Print(StringFormatted("[WARNING] Unknown attribute {}\n", name));
```

File: tst/dot.tst [hunk 2]
```diff
@@ -65,5 +65,18 @@ gap> g := GraphvizGraph();;
 gap> AsString(g);
 "graph  {\n}\n"
 
+# Test unknown attributes (node)
+gap> g := GraphvizGraph();;
+gap> n := GraphvizAddNode(g, "n");;
+gap> GraphvizSetAttr(n, "test", "false");
+[WARNING] Unkown attribute test
```
Comment (james-d-mitchell) [inline tst/dot.tst:72]:
```suggestion
[WARNING] Unknown attribute test
```

File: tst/dot.tst [hunk 3]
```diff
@@ -65,5 +65,18 @@ gap> g := GraphvizGraph();;
...
+<node n>
+
+# Test unknown attributes (graph)
+gap> g := GraphvizGraph();;
+gap> GraphvizSetAttr(g, "test", "false");
+[WARNING] Unkown attribute test
```
Comment (james-d-mitchell) [inline tst/dot.tst:78]:
```suggestion
[WARNING] Unknown attribute test
```

**Comment by mtorpey** [comment]
Instead of merging, I've squashed this branch and pushed it directly to `main`, so this is all in the main branch now.

I'll close this PR!

**Pull request closed without merging**